lib/SILOptimizer/SILCombiner/SILCombiner.h

```diff
@@ -295,6 +295,16 @@ class SILCombiner :
 private:
   FullApplySite rewriteApplyCallee(FullApplySite apply, SILValue callee);
 
+  Optional<ConcreteExistentialInfo>
+  buildConcreteExistentialInfo(Operand &ArgOperand);
```

Comment:
Good point, however, please keep in mind that we then need to explicitly pass ProtocolConformanceAnalysis and ClassHierarchyAnalysis as arguments to these static APIs.

Reply:
Also, buildConcreteExistentialInfoUsingSoleConformingType needs SIlBuilder to append instructions. So it may make more sense to keep it the way it is right now.